src/main/java/com/princexml/wrapper/PrinceControl.java

```diff
@@ -111,7 +111,9 @@ private boolean convert(OutputStream output) throws IOException {
         if (chunk.getTag().equals("log")) {
             BufferedReader br = new BufferedReader(new InputStreamReader(
                     new ByteArrayInputStream(chunk.getBytes())));
-            return readMessages(br);
+            boolean result = readMessages(br);
+            br.close();
```

Comment:
Is this close necessary given it's just reading from an array?

Reply:
Good catch, thanks. Closing a `ByteArrayInputStream` does nothing at all.

Reply:
For the sake of consistency, I've enclosed these in try-with-resources statements too. Just so that down the track, I don't have to remember if I've intentionally not closed the stream or if I actually forgot.